src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.Remote.cs

```diff
@@ -1,38 +1,44 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Experiments;
 using Microsoft.CodeAnalysis.Remote;
 
 namespace Microsoft.CodeAnalysis.NavigateTo
 {
     internal abstract partial class AbstractNavigateToSearchService
     {
         private async Task<ImmutableArray<INavigateToSearchResult>> SearchDocumentInRemoteProcessAsync(
-            RemoteHostClient.Session session, Document document, string searchPattern, CancellationToken cancellationToken)
+            RemoteHostClient client, Document document, string searchPattern, CancellationToken cancellationToken)
         {
-            var serializableResults = await session.InvokeAsync<ImmutableArray<SerializableNavigateToSearchResult>>(
-                nameof(IRemoteNavigateToSearchService.SearchDocumentAsync),
-                document.Id, searchPattern).ConfigureAwait(false);
+            var solution = document.Project.Solution;
 
-            return serializableResults.SelectAsArray(r => r.Rehydrate(document.Project.Solution));
+            var serializableResults = await client.RunCodeAnalysisServiceOnRemoteHostAsync<ImmutableArray<SerializableNavigateToSearchResult>>(
+                solution, nameof(IRemoteNavigateToSearchService.SearchDocumentAsync),
+                new object[] { document.Id, searchPattern }, cancellationToken).ConfigureAwait(false);
+
+            return serializableResults.NullToEmpty().SelectAsArray(r => r.Rehydrate(solution));
```

Comment:
Here you use NullToEmpty (which i like), but in the code a few files up, you have a "bool + default-arrray" and you have to check the bool.  Why the different patterns?